src/evo/mnhftx.cpp

```diff
@@ -13,29 +13,39 @@
 #include <chain.h>
 #include <chainparams.h>
 #include <validation.h>
+#include <versionbits.h>
 
 #include <string>
 
-extern const std::string CBLSIG_REQUESTID_PREFIX = "clsig";
+extern const std::string MNEHF_REQUESTID_PREFIX = "mnhf";
 
-bool MNHFTx::Verify(const CBlockIndex* pQuorumIndex) const
+bool MNHFTx::Verify(const CBlockIndex* pQuorumIndex, const uint256& msgHash, TxValidationState& state) const
 {
-    if (nVersion == 0 || nVersion > (llmq::utils::IsV19Active(pQuorumIndex) ? BASIC_BLS_VERSION : LEGACY_BLS_VERSION)) {
-        return false;
+    if (versionBit >= VERSIONBITS_NUM_BITS) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-mnhf-nbit-out-of-bounds");
     }
 
     Consensus::LLMQType llmqType = Params().GetConsensus().llmqTypeMnhf;
     const auto& llmq_params_opt = llmq::GetLLMQParams(llmqType);
-    assert(llmq_params_opt.has_value());
+    if (!llmq_params_opt.has_value()) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-mnhf-quorum-type");
+    }
     int signOffset{llmq_params_opt->dkgInterval};
 
-    const uint256 requestId = ::SerializeHash(std::make_pair(CBLSIG_REQUESTID_PREFIX, pQuorumIndex->nHeight));
-    return llmq::CSigningManager::VerifyRecoveredSig(llmqType, *llmq::quorumManager, pQuorumIndex->nHeight, requestId, pQuorumIndex->GetBlockHash(), sig, 0) ||
-           llmq::CSigningManager::VerifyRecoveredSig(llmqType, *llmq::quorumManager, pQuorumIndex->nHeight, requestId, pQuorumIndex->GetBlockHash(), sig, signOffset);
+    const uint256 requestId = ::SerializeHash(std::make_pair(MNEHF_REQUESTID_PREFIX, static_cast<int64_t>(versionBit)));
```

Comment:
```suggestion
    const uint256 requestId = ::SerializeHash(std::make_pair(MNEHF_REQUESTID_PREFIX, int64_t{versionBit}));
```

Reply:
why is it better except 10 characters shorter?